### PR TITLE
New header "X-Load-Balancing-Endpoint-Weight" returned by service health

### PR DESCRIPTION
**Pull request by cezarygerard**
add new header "X-Load-Balancing-Endpoint-Weight" returned from service health. Value of the header is number of local endpoints. Header can be used in weighted load balancing. Parsing header for number of endpoints is faster than unmarshalling json from the content body, and it can be returned via HTTP HEAD method.

The header has been already added to kube-proxy service health code in https://github.com/kubernetes/kubernetes/pull/118999 (will be released in kubernetes v1.28)

```release-note
For services with `External Traffic Policy: Local` Service health returns http header "X-Load-Balancing-Endpoint-Weight" with number of local endpoints. The same information is still available in response body JSON payload.LocalEndpoints.
```

### CODE REVIEW

**Review by alan-kut — APPROVED** [review]

**Review by ysksuzuki — APPROVED** [review]

**Comment by cezarygerard** [comment]
@ysksuzuki @joestringer @alan-kut 

```

Merging is blocked
The base branch restricts merging to authorized users.
```

Who should  be able to merge this branch?

**Comment by joestringer** [comment]
All tests must pass and all codeowner reviews must be checked off, then typically there's a rotating hat for a committer to sweep through and merge PRs like this. I've kicked off the full testsuite to confirm the other requirement before merge.

**Comment by youngnick** [comment]
Going to give the tests one more crack, I've seen some weird connectivity shenanigans in the last 24 hours.

**Pull request merged**
